metabrainz/__init__.py

```diff
@@ -10,21 +15,30 @@ def create_app(config_path=None):
         use_debug_toolbar=True,
     )
 
-    # Configuration files
-    app.config.from_pyfile(os.path.join(
-        os.path.dirname(os.path.realpath(__file__)),
-        '..', 'default_config.py'
-    ))
-    app.config.from_pyfile(os.path.join(
-        os.path.dirname(os.path.realpath(__file__)),
-        '..', 'consul_config.py'
-    ), silent=True)
-    app.config.from_pyfile(os.path.join(
-        os.path.dirname(os.path.realpath(__file__)),
-        '..', 'custom_config.py'
-    ), silent=True)
-    if config_path:
-        app.config.from_pyfile(config_path)
+    # Load configuration files
+    # If we're running under a docker deployment, only use the consul configuration.
+    if deploy_env:
+        print("Running in production!");
+        consul_config = os.path.join( os.path.dirname(os.path.realpath(__file__)), 
+            '..', 'consul_config.py')
+        while not os.path.exists(consul_config):
+            print("No configuration file generated yet. Sleeping...");
+            sleep(2)
+
+        app.config.from_pyfile(consul_config, silent=True)
+
+    else:
+        # Otherwise, do our best to find configu files
+        app.config.from_pyfile(os.path.join(
+            os.path.dirname(os.path.realpath(__file__)),
+            '..', 'default_config.py'
```

Comment:
I don't know how config files in production work here - does consul_config contain all settings that exist in default_config, or does it just override a few things? If it overrides only some settings you should always do this one, and then optionally  load consul/custom config in your if block

Reply:
It provides everything needed to run the service. This is the template file that is filled out with actual values:

https://github.com/metabrainz/metabrainz.org/blob/master/consul_config.py.ctmpl

No other config files should be used or values overridden.

Reply:
There are many items in default_config that don't exist in the consul_config template:
https://github.com/metabrainz/metabrainz.org/blob/7738c77b7ffb4eac336a399878b28255dbd79fe9/default_config.py#L16-L17

Reply:
Ugh. Good catch! That makes this solution a lot less clean. :(